PocketKit/Sources/PocketKit/PocketNotificationService.swift

```diff
@@ -0,0 +1,217 @@
+        center.requestAuthorization(options: [.badge, .sound, .alert, .provisional]) { granted, error in
+          print("Notification authorization, granted: \(granted), error: \(String(describing: error))")
+        }
+    }
+    
+    /**
+     Proxy function to braze to register a device token
+     */
+    public func register(deviceToken: Data) {
+        //TODO: Also register the token with the Pocket Instant Sync endpoint
+        self.braze.notifications.register(deviceToken: deviceToken)
+    }
+    
+    /**
+     Proxy function to braze to handle background notifications
+     */
+    public func handleBackgroundNotifcation(didReceiveRemoteNotification userInfo: [AnyHashable : Any],
+                                  fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) {
+              
+        // Forward background notification to Braze.
+        if braze.notifications.handleBackgroundNotification(
+          userInfo: userInfo,
+          fetchCompletionHandler: completionHandler
+        ) {
+         // Braze handled the notification, nothing more to do.
+         return
+       }
+        
+        
+        // Braze did not handle this remote background notification.
+        // We can handle the notification yourself here.
+        //TODO: Handle the V3 Instant sync push notification.
+
+        // Manually call the completion handler to let the system know
+        // that the background notification is processed.
+        completionHandler(.noData)
+    }
+    
+    /**
+     Proxy function to Braze to handle user facing notifications
+     */
+    public func didRecieveUserNotifcation( _ center: UNUserNotificationCenter,
+                            didReceive response: UNNotificationResponse,
+        withCompletionHandler completionHandler: @escaping () -> Void) {
+        let _ = self.braze.notifications.handleUserNotification(response: response, withCompletionHandler: completionHandler)
+    }
+}
+
+/**
+ Extend the Notification service to support User Notifications
+ */
+extension PocketNotificationService : UNUserNotificationCenterDelegate {
+   
+    func userNotificationCenter(
+                                      _ center: UNUserNotificationCenter,
+                           didReceive response: UNNotificationResponse,
+       withCompletionHandler completionHandler: @escaping () -> Void
+     ) {
+         self.didRecieveUserNotifcation(center, didReceive: response, withCompletionHandler: completionHandler)
+     }
+
+
+     func userNotificationCenter(
+                                      _ center: UNUserNotificationCenter,
+                      willPresent notification: UNNotification,
+       withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void
+     ) {
+         completionHandler([.list, .banner])
+     }
+}
+
+/**
+ Extend the App Delegate so that we can add in the notification methods
+ */
+extension PocketAppDelegate {
```

Comment:
I like having this code in an extension but would prefer to have it either 1) within the actual `PocketAppDelegate.swift` file or in another file following the naming convention of `PocketAppDelegate+foo.swift`.